From 07f4dd5c860a40cc46f839c57bf1751ed9ce1c2e Mon Sep 17 00:00:00 2001
From: Mike <mike@example.com>
Date: Sat, 22 Feb 2026 17:30:00 +0000
Subject: [PATCH] Encode: disable recursive subdir builds

---
 ext/Encode/Makefile.PL | 1 +
 1 file changed, 1 insertion(+)

diff --git perl-5.8.9/ext/Encode/Makefile.PL perl-5.8.9/ext/Encode/Makefile.PL
index f5ef65e..b71baf5 100644
--- perl-5.8.9/ext/Encode/Makefile.PL
+++ perl-5.8.9/ext/Encode/Makefile.PL
@@ -43,6 +43,7 @@ WriteMakefile(
     @man,
     INC         => '-I' . File::Spec->catfile( '.', 'Encode' ),
     PMLIBDIRS   => \@pmlibdirs,
+    DIR         => [],
     INSTALLDIRS => 'perl',
 );
 
-- 
2.49.1
